core/src/main/java/org/elasticsearch/search/aggregations/support/InternalOrder.java

```diff
@@ -327,40 +342,82 @@ public static void writeOrder(Terms.Order order, StreamOutput out) throws IOExce
                 out.writeString(path.toString());
             } else if (order instanceof CompoundOrder) {
                 CompoundOrder compoundOrder = (CompoundOrder) order;
-                    out.writeByte(order.id());
-                    out.writeVInt(compoundOrder.orderElements.size());
-                    for (Terms.Order innerOrder : compoundOrder.orderElements) {
-                        Streams.writeOrder(innerOrder, out);
-                    }
+                out.writeByte(order.id());
+                out.writeVInt(compoundOrder.orderElements.size());
+                for (BucketOrder innerOrder : compoundOrder.orderElements) {
+                    Streams.writeOrder(innerOrder, out);
+                }
             } else {
                 out.writeByte(order.id());
             }
         }
 
-        public static Terms.Order readOrder(StreamInput in) throws IOException {
-            return readOrder(in, false);
-        }
-
-        public static Terms.Order readOrder(StreamInput in, boolean absoluteOrder) throws IOException {
+        public static BucketOrder readOrder(StreamInput in) throws IOException {
+            // TODO backwards compat
```

Comment:
@colings86 made the changes here: https://github.com/elastic/elasticsearch/pull/22343/commits/fa54c8e037017bb9826082e59d0f741bfa300ebe
See this line for the `writeTo` backward compat logic that I was not sure about: https://github.com/elastic/elasticsearch/pull/22343/commits/fa54c8e037017bb9826082e59d0f741bfa300ebe#diff-78913b4f2494b84e11cc7cc3e12c5648R349
Still need to update documentation for the breaking Java API changes.